monitored-httpclient/src/test/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/HttpClientPoolTest.java

```diff
@@ -141,16 +141,16 @@ void shouldUseNullTruststore() throws MalformedURLException, URISyntaxException
     }
 
     @Test
-    void whenCreatingAnHttpClientPoolWithAValidHostnameThenThePoolShoudBeNonEmptyAndHealthy() {
+    void shouldReturnToString() {
         var hostname = "google.com";
         final ServerConfiguration serverConfiguration = new ServerConfiguration(hostname);
         try (HttpClientPool httpClientPool = HttpClientPool.builder(serverConfiguration)
                 .withScheduledExecutorService(Executors.newScheduledThreadPool(4))
                 .build()
         ) {
             assertFalse(httpClientPool.getNextHttpClient().isEmpty());
-            assertTrue(httpClientPool.getNextHttpClient().get().isHealthy());
```

Comment:
it is not possible to get the pool unhealthy by mocking the checkHealth?

Reply:
I think yes, but In this case we should use a spy on HttpClientPool.